projects/observability/src/pages/explorer/explorer-service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable } from '@angular/core';
+import { NavigationParams, NavigationParamsType } from '@hypertrace/common';
+import { Filter, FilterBuilderLookupService } from '@hypertrace/components';
+import { MetadataService, SPAN_SCOPE, toFilterAttributeType } from '@hypertrace/distributed-tracing';
+import { combineLatest, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { ObservabilityTraceType } from '../../shared/graphql/model/schema/observability-traces';
+import { ScopeQueryParam } from './explorer.component';
+
+@Injectable({ providedIn: 'root' })
+export class ExplorerService {
+  public constructor(
+    private readonly metadataService: MetadataService,
+    private readonly filterBuilderLookupService: FilterBuilderLookupService
+  ) {}
+  public buildNavParamsWithFilters(
+    scopeQueryParam: ScopeQueryParam,
+    filters: DrilldownFilter[]
+  ): Observable<NavigationParams> {
+    const filterStrings$: Observable<string>[] = filters.map(filter =>
+      this.metadataService
+        .getAttribute(
+          scopeQueryParam === ScopeQueryParam.EndpointTraces ? ObservabilityTraceType.Api : SPAN_SCOPE,
+          filter.field
+        )
+        .pipe(
+          map(attribute => ({ ...attribute, type: toFilterAttributeType(attribute.type) })),
+          map(
+            filterAttribute =>
+              this.filterBuilderLookupService
+                .lookup(filterAttribute.type)
+                .buildFilter(filterAttribute, filter.operator, filter.value).urlString
+          )
+        )
+    );
+
+    return combineLatest(filterStrings$).pipe(
```

Comment:
Although functionally the same here, I think forkJoin is a better semantic fit here which conveys how this is meant to work. Basically:

forkJoin - When all observables complete, emit the last emitted value from each.

combineLatest - When any observable emits a value, emit the latest value from each.